--- filters.js	2022-08-05 01:24:21.753247500 +0800
+++ filters-new.js	2022-08-05 01:22:50.953605900 +0800
@@ -39,7 +39,13 @@
     for (const module of await fs_1.promises.readdir(config_1.config_filter_path)) {
         const modulePath = (0, path_1.join)(config_1.config_filter_path, module);
         debug(`loading filter ${module} of ${modulePath}`);
-        const { default: filter_function } = await Promise.resolve().then(() => __importStar(require(modulePath)));
+        let filter_function;
+        if(modulePath.endsWith('.mjs')){
+            filter_function = (await Promise.resolve().then(() => import(`file://${modulePath}`))).default;
+        }
+        else if(modulePath.endsWith('.js')){
+            filter_function = (await Promise.resolve().then(() => __importStar(require(modulePath)))).default;
+        }
         debug(`filter_function ${filter_function}, typeof ${typeof filter_function}`);
         yield filter_function;
     }
